src/functions-and-arrays.js

```diff
@@ -1,43 +1,165 @@
-// Iteration #1: Find the maximum
-function maxOfTwoNumbers() {}
-
+// Iteration #1: Find the maximum 
+// Implement the function maxOfTwoNumbers that takes two numbers as arguments and returns the bigger number.
+function maxOfTwoNumbers(num1, num2) {
+  if(num1 > num2){
+    return num1
+  }
+  else if (num2 > num1){
+    return num2
+  }
+  else {
+    return num1, num2;
+  }
+}
 
 
 // Iteration #2: Find longest word
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
 
-function findLongestWord() {}
-
+function findLongestWord(arr) {
+  
+  if(arr.length === 0){
+    return null;
+  }
+  else if (arr.length === 1){
+    return arr[0];
+  }
+
+  let largest = 0;
+  let largestWord;
+  for (let i = 0; i < arr.length; i++){
+    
+    if(arr[i].length>largest){
+      largest = arr[i].length;
+      largestWord = arr[i];
+    }
+    else if(arr[i].length === largest){
+      continue;
+    }
+  
+  }
+  return largestWord;
+  
+}
 
+ 
 
 // Iteration #3: Calculate the sum
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
-function sumNumbers() {}
-
 
+function sumNumbers(array){
+  
+  if(array.length === 0){
+    return 0
+  }
+  if(array.length === 1){
+    return array[0];
+  }
+  let sum = 0;
+  array.forEach(element => {
+    sum += element;
+  });
+  return sum 
+  
+}
 
 // Iteration #3.1 Bonus:
-function sum() {}
+
+function sum(array) {
+  if(array.length === 0){
+    return 0;
+  }
+  let sum = 0;
+  for(let i = 0; i < array.length; i++){
+    if(typeof array[i] === "boolean" && array[i]){
+      sum += 1;
+      continue;
+    }
+    else if(typeof array[i] === "string" && array[i].length > 0){
+      sum += array[i].length;
+      continue;
+      
+    }
+    else if(typeof array[i] === "number"){
+      sum += array[i];
+      continue;
+    }
+    else if(Array.isArray(array[i]) || typeof array[i] === "object"){
+      throw new Error("Unsupported data type sir or ma'am");
+    }
+    
+  }
+  return sum
+}
 
 
 
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
-function averageNumbers() {}
+function averageNumbers(arr) {
+  let result;
+  if (arr.length === 0){
+    result = null;
+  }
+  arr.forEach(element => {
```

Comment:
Filipa, here, since you are reusing your `sumNumbers` function, you don't need to run a loop over the argument of the `averageNumbers` function as the `sumNumbers` will do that for you. So, in this case, you could simplify your answer by just this:
```
function averageNumbers(arr) {
  if (!arr.length) return null;

  let sum = sumNumbers(arr);
  let result = sum / arr.length;

  return result;
}
```